src/OpenTelemetry.AutoInstrumentation.BuildTasks/CheckForInstrumentationPackages.cs

```diff
@@ -0,0 +1,161 @@
+    /// Gets or sets the list of instrumentation target items.
+    /// </summary>
+    /// <remarks>
+    /// See the <code>InstrumentationTarget</code>"/> items on the <code>.targets</code> file for a list of the
+    /// required metadata.
+    /// </remarks>
+    [Required]
+    public ITaskItem[]? InstrumentationTargetItems { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of runtime copy local items.
+    /// </summary>
+    /// <remarks>
+    /// This list generated by the standard build target <code>ResolvePackageAssets</code>.
+    /// </remarks>
+    [Required]
+    public ITaskItem[]? RuntimeCopyLocalItems { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to use verbose logging.
+    /// </summary>
+    public bool UseVerboseLog { get; set; }
+
+    /// <summary>
+    /// Checks if the project references any packages that can be instrumented and if the respective adapter packages
+    /// were already added to the project.
+    /// </summary>
+    /// <returns>True, if the task completed without errors, false otherwise.</returns>
+    public override bool Execute()
+    {
+        _logMessageImportance = UseVerboseLog ? MessageImportance.High : MessageImportance.Low;
+        try
+        {
+            return ExecuteImplementation();
+        }
+        catch (Exception ex)
+        {
+            Log.LogError($"{LogPrefix}{ex}");
+            throw;
+        }
+    }
+
+    private bool ExecuteImplementation()
+    {
+        if (RuntimeCopyLocalItems is null || RuntimeCopyLocalItems.Length == 0)
+        {
+            Log.LogMessage(
+                _logMessageImportance,
+                $"{LogPrefix}empty {nameof(RuntimeCopyLocalItems)}, skipping check for instrumentation packages.");
+            return true;
+        }
+
+        // Put runtime local items in a dictionary for easy lookup.
+        var runtimeCopyLocalItemsDictionary = new Dictionary<string, ITaskItem>(
+            RuntimeCopyLocalItems.Length, StringComparer.OrdinalIgnoreCase);
+        foreach (var item in RuntimeCopyLocalItems)
+        {
+            var nuGetPackageId = item.GetMetadata("NuGetPackageId");
+            if (!string.IsNullOrWhiteSpace(nuGetPackageId))
+            {
+                runtimeCopyLocalItemsDictionary[nuGetPackageId] = item;
+            }
+        }
+
+        var missingAnyInstrumentationAdapterPackage = false;
+
+        // Process each instrumentation target item.
+        foreach (var instrumentationTarget in InstrumentationTargetItems ?? Array.Empty<ITaskItem>())
+        {
+            missingAnyInstrumentationAdapterPackage |=
+                MissingInstrumentationAdapterPackage(instrumentationTarget, runtimeCopyLocalItemsDictionary);
+        }
+
+        // If missing any instrumentation adapter package then fail the build by returning false, the method
+        // Log.LogError was already called.
+        return !missingAnyInstrumentationAdapterPackage;
```

Comment:
I think instead of tracking it manually you could simply return `Log.HasLoggedErrors`

Reply:
d1b9876d7df02e869fa748fa3e19275c71d27865